map/generate-offline-map-overrides/src/main/java/com/esri/samples/generate_offline_map_overrides/GenerateOfflineMapOverridesController.java

```diff
@@ -123,122 +119,117 @@ private void initialize() {
    */
   @FXML
   private void generateOfflineMap() {
-    try {
-      // show the progress bar
-      progressBar.setVisible(true);
+    // show the progress bar
+    progressBar.setVisible(true);
 
-      // create an offline map task with the map
-      var offlineMapTask = new OfflineMapTask(map);
-
-      // get default offline map parameters for this task given the download area
-      ListenableFuture<GenerateOfflineMapParameters> generateOfflineMapParametersFuture = offlineMapTask
-        .createDefaultGenerateOfflineMapParametersAsync(downloadArea.getGeometry());
-
-      generateOfflineMapParametersFuture.addDoneListener(() -> {
-        try {
-          final GenerateOfflineMapParameters parameters = generateOfflineMapParametersFuture.get();
+    // create an offline map task with the map
+    var offlineMapTask = new OfflineMapTask(map);
 
+    // get default offline map parameters for this task given the download area
+    offlineMapTask.createDefaultGenerateOfflineMapParametersAsync(downloadArea.getGeometry())
+      .toCompletableFuture().whenComplete((parameters, ex) -> {
```

Comment:
I can see the benefits here where we've got more bracket indentations going on, of being able to see both the exception actions and the successful actions without needing to find the right bracket further down. Assuming I've got the approach right here:
```java
offlineMapTask.createGenerateOfflineMapParameterOverridesAsync(parameters).toCompletableFuture().exceptionally(ex -> {
             // if the operation completes with an exception, display an alert
              new Alert(Alert.AlertType.ERROR, "Error creating override parameters: " + e.getMessage()).show();
              return null;
            }).thenAccept(overrides -> {
             // if the operation completes successfully
              try {
                // get the export tile cache parameters for the base layer
                var basemapParamKey = new OfflineMapParametersKey(
                    mapView.getMap().getBasemap().getBaseLayers().get(0));
                ExportTileCacheParameters exportTileCacheParameters =
                    overrides.getExportTileCacheParameters().get(basemapParamKey);
// etc etc etc
```

Returning null feels a bit weird but as far as I can tell it's what you'd do? Any additional thoughts on benefits of the approach above (if it's valid) vs using `whenComplete` here @colinanderson ? Is it purely readability side I noted, or are there other potential benefits?

Reply:
Probably `exceptionally` works better when you can provide a default value and `null` is probably not a good value. `whenComplete` is ok here. If the future could return a empty list for example then exceptionally might be cleaner looking because it could return an empty list and the `thenApply` would work as normal - assuming it already handled empty list which it should.